src/navier_stokes/INSStaggeredHierarchyIntegrator.cpp

```diff
@@ -939,47 +958,21 @@ INSStaggeredHierarchyIntegrator::initializeHierarchyIntegrator(Pointer<PatchHier
         d_F_div_idx = -1;
     }
 
-    // Register variables for tracking mean flow quantities and turbulent kinetic
-    // energy.
-    if (d_U_mean_var)
+    // Register variables for tracking mean flow quantities.
+    d_U_mean_current_idxs.resize(d_num_flow_snapshots, IBTK::invalid_index);
+    d_U_mean_new_idxs.resize(d_num_flow_snapshots, IBTK::invalid_index);
+    d_U_mean_scratch_idxs.resize(d_num_flow_snapshots, IBTK::invalid_index);
+    for (unsigned int k = 0; k < d_num_flow_snapshots; ++k)
     {
-        registerVariable(d_U_mean_current_idx,
-                         d_U_mean_new_idx,
-                         d_U_mean_scratch_idx,
-                         d_U_mean_var,
-                         cell_ghosts,
+        registerVariable(d_U_mean_current_idxs[k],
+                         d_U_mean_new_idxs[k],
+                         d_U_mean_scratch_idxs[k],
+                         d_U_mean_vars[k],
+                         side_ghosts,
                          d_U_mean_coarsen_type,
                          d_U_mean_refine_type);
```

Comment:
I suppose that this approach is necessary but it means that regrids will continue to get slower and slower since every snapshot will be regridded.